### PR TITLE
dynamic shapes: support randn.generator when generator is None

### PR DESCRIPTION
**Pull request by bdhirsh**
Fixes https://github.com/pytorch/pytorch/issues/107170

Today, we have a decomp for `randn.default` that handles dynamic shapes properly.

It looks like we don't have a decomp for the `randn.generator` overload of randn - but if you pass in a `None` generator, we can just call the default decomp. We should probably support the generator overload at some point (although I'm not sure what we do with generators in our IR today).

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #116223

### CODE REVIEW

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -5427,6 +5427,32 @@ def randn(
...
+    generator: Optional[torch.Generator] = None,
+    dtype: Optional[torch.dtype] = None,
+    device: Optional[DeviceLikeType] = None,
+    layout: Optional[torch.layout] = None,
+    requires_grad: bool = False,
+    pin_memory: bool = False,
+) -> TensorLikeType:
+    # We should eventually support the generator overload.
+    # However, if someone passes in a None generator explicitly,
+    # we can jut fall back to randn.default
```
Comment (Skylion007) [inline torch/_refs/__init__.py:5443]:
```suggestion
    # we can just fall back to randn.default
```

**Review by Skylion007 — APPROVED** [review]

**Comment by bdhirsh** [comment]
Need to investigate CI (it roughly looks like dynamo isn't properly graph breaking on `randn.Generator` anymore when a generator is actually passed in)

**Comment by aviator19941** [comment]
@bdhirsh is there an update on merging this in? I've been keeping a manual aten.randn.generator decomposition in my codebase in the meanwhile, but was wondering if this was going to get merged in. Thank you very much!

**Pull request closed without merging**